src/marqo/tensor_search/add_docs.py

```diff
@@ -127,27 +127,42 @@ def threaded_download_and_preprocess_content(allocated_docs: List[dict],
-                            ModelType.LanguageBind] and modality not in marqo_index.model.properties.get(
+                            ModelType.LanguageBind] and inferred_modality not in marqo_index.model.properties.get(
                             'supported_modalities'):
                             media_repo[doc[field]] = UnsupportedModalityError(
-                                f"Model {marqo_index.model.name} does not support {modality}")
+                                f"Model {marqo_index.model.name} does not support {inferred_modality}")
                             continue
+
+                        if is_structured_index:
+                            if inferred_modality is Modality.VIDEO and media_field_types_mapping[
+                            field] is not FieldType.VideoPointer:
+                                media_repo[doc[field]] = S2InferenceError(
+                                    f"Invalid audio file. Error processing media file {doc}, detected as video, but field type is not VideoPointer")
+                                continue
+
+                            if inferred_modality is Modality.AUDIO and media_field_types_mapping[
+                            field] is not FieldType.AudioPointer:
+                                media_repo[doc[field]] = S2InferenceError(
+                                    f"Invalid video file. Error processing media file {doc}, detected as audio, but field type is not AudioPointer")
+                                continue
+
                         try:
-                            processed_chunks = download_and_chunk_media(doc[field], device, download_headers, modality,
-                                                                        marqo_index, preprocessors)
+                            processed_chunks = download_and_chunk_media(doc[field], device, download_headers, inferred_modality,
+                                                                    marqo_index, preprocessors)
                             media_repo[doc[field]] = processed_chunks
                         except (ffmpeg.Error, S2InferenceError) as e:
-                            logger.error(f"Error processing {modality} file: {str(e)}")
-                            media_repo[doc[field]] = S2InferenceError(f"Error processing {modality} file: {str(e)}")
-
+                            logger.error(f"Error processing {inferred_modality} file: {str(e)}")
+                            media_repo[doc[field]] = S2InferenceError(f"Error processing {inferred_modality} file: {str(e)}")
+                    
+                    elif inferred_modality is Modality.TEXT and is_structured_index and media_field_types_mapping[field] in [FieldType.AudioPointer, FieldType.VideoPointer, FieldType.ImagePointer]:
+                        media_repo[doc[field]] = S2InferenceError(f"Error processing media file {doc}, detected as text or failed to determine modality")
```

Comment:
@farshidz 
If we had a network issue when inferring the modality, it would not be included in the error message to the user. Working on this will require extra time. Is this generic error message acceptable?

Reply:
Done, changed to pass the error to doc